velox/dwio/parquet/reader/TimestampColumnReader.h

```diff
@@ -82,4 +70,186 @@ class TimestampColumnReader : public IntegerColumnReader {
+          addOutputRow(rows[i]);
+          idx++;
+        }
+      } else {
+        if (!bits::isBitNull(rawNulls, i)) {
+          bits::setNull(rawResultNulls_, idx, false);
+          rawTs[idx] = rawTs[i];
+          addOutputRow(rows[i]);
+          idx++;
+        }
+      }
+    }
+  }
+
+  void processFilter(
+      const common::Filter* filter,
+      const RowSet rows,
+      const uint64_t* rawNulls) {
+    auto rawTs = values_->asMutable<Timestamp>();
+
+    returnReaderNulls_ = false;
+    anyNulls_ = false;
+    allNull_ = true;
+    vector_size_t idx = 0;
+    for (vector_size_t i = 0; i < numValues_; i++) {
+      if (rawNulls && bits::isBitNull(rawNulls, i)) {
+        if (filter->testNull()) {
+          bits::setNull(rawResultNulls_, idx);
+          addOutputRow(rows[i]);
+          anyNulls_ = true;
+          idx++;
+        }
+      } else {
+        if (filter->testTimestamp(rawTs[i])) {
+          if (rawNulls) {
+            bits::setNull(rawResultNulls_, idx, false);
+          }
+          rawTs[idx] = rawTs[i];
+          addOutputRow(rows[i]);
+          allNull_ = false;
+          idx++;
+        }
+      }
+    }
+  }
+
+  void getValues(RowSet rows, VectorPtr* result) override {
+    getFlatValues<Timestamp, Timestamp>(rows, result, requestedType_);
+  }
+
+  template <bool isDense>
+  void readHelper(common::Filter* filter, RowSet rows) {
+    dwio::common::ExtractToReader extractValues(this);
+    common::AlwaysTrue alwaysTrue;
+    dwio::common::ColumnVisitor<
+        int64_t,
+        common::AlwaysTrue,
+        decltype(extractValues),
+        isDense>
+        visitor(alwaysTrue, this, rows, extractValues);
+    readWithVisitor(rows, visitor);
+
+    auto tsValues =
+        AlignedBuffer::allocate<Timestamp>(numValues_, &memoryPool_);
+    auto rawTs = tsValues->asMutable<Timestamp>();
+    auto rawTsInt64 = values_->asMutable<int64_t>();
+
+    const auto rawNulls =
+        resultNulls() ? resultNulls()->as<uint64_t>() : nullptr;
+
+    Timestamp timestamp;
+    for (vector_size_t i = 0; i < numValues_; i++) {
+      if (!rawNulls || !bits::isBitNull(rawNulls, i)) {
+        if (parquetTimestampPrecision_ == TimestampPrecision::kMicroseconds) {
+          timestamp = Timestamp::fromMicros(rawTsInt64[i]);
+        } else {
+          timestamp = Timestamp::fromMillis(rawTsInt64[i]);
+        }
+        rawTs[i] = timestamp;
+        rawTs[i].toPrecision(timestampPrecision_);
+      }
+    }
+
+    values_ = tsValues;
+    rawValues_ = values_->asMutable<char>();
+    outputRows_.clear();
+
+    switch (
+        !filter ||
+                (filter->kind() == common::FilterKind::kIsNotNull && !rawNulls)
+            ? common::FilterKind::kAlwaysTrue
+            : filter->kind()) {
+      case common::FilterKind::kAlwaysTrue:
+        // Simply add all rows to output.
+        for (vector_size_t i = 0; i < numValues_; i++) {
+          addOutputRow(rows[i]);
```

Comment:
I don't think you need to do this, we use `inputRows_` in case there is no filter